src/makeCalculator.js

```diff
@@ -1,10 +1,39 @@
 'use strict';
 
-/**
- * @return {object}
- */
 function makeCalculator() {
-  // write code here
+  const resultObj = {
+    result: 0,
+
+    add(number) {
+      resultObj.result += number;
+    },
+
+    subtract(number) {
+      resultObj.result -= number;
+    },
+
+    multiply(number) {
+      resultObj.result *= number;
+    },
+
+    divide(number) {
+      resultObj.result /= number;
+    },
+
+    operate(callback, number) {
+      callback(number);
+
+      return this;
+    },
+
+    reset() {
+      resultObj.result = 0;
+
+      return this;
+    },
+  };
```

Comment:
[NAMING] - The variable `resultObj` should be named according to the function `makeCalculator`. A more appropriate name could be `calculator` to reflect the object it represents and to comply with the checklist item 1.